src/main/java/dev/galacticraft/mod/content/block/special/fluidpipe/GlassFluidPipeBlock.java

```diff
@@ -125,14 +131,21 @@ public InteractionResult use(BlockState blockState, Level level, BlockPos blockP
     @Override
     public void neighborChanged(BlockState blockState, Level level, BlockPos blockPos, Block block, BlockPos fromPos, boolean notify) {
         super.neighborChanged(blockState, level, blockPos, block, fromPos, notify);
-        var neighborState = level.getBlockState(fromPos);
-        var delta = fromPos.subtract(blockPos);
-        var direction = Direction.fromDelta(delta.getX(), delta.getY(), delta.getZ());
+        System.out.println("neighborChanged");
 
-        if (direction != null && level.getBlockEntity(blockPos) instanceof GlassFluidPipeBlockEntity glassPipe) {
-            var otherBlockEntity = level.getBlockEntity(fromPos);
-            glassPipe.getConnections()[direction.ordinal()] = !neighborState.isAir() && ((otherBlockEntity instanceof Pipe pipe && pipe.canConnect(direction.getOpposite())) || FluidUtil.canAccessFluid(level, fromPos, direction));
-            level.sendBlockUpdated(blockPos, blockState, blockState, Block.UPDATE_IMMEDIATE);
+        if (level.getBlockEntity(blockPos) instanceof PipeBlockEntity glassPipe) {
+            var direction = DirectionUtil.fromNormal(fromPos.getX() - blockPos.getX(), fromPos.getY() - blockPos.getY(), fromPos.getZ() - blockPos.getZ());
+
+            if (direction != null) {
+                if (!level.isClientSide
+                        && glassPipe.getConnections()[direction.ordinal()]
+                        != (glassPipe.getConnections()[direction.ordinal()]
+                        = glassPipe.canConnect(direction) && FluidUtil.canAccessFluid(level, fromPos, direction))
+                ) {
+                    level.sendBlockUpdated(blockPos, blockState, blockState, Block.UPDATE_IMMEDIATE);
+                }
+            }
+            glassPipe.setChanged();
```

Comment:
Same as the other `sendBlockUpdated`/`setChanged` pair, but the other way around.

Reply:
Done